ruby/vampires.rb

```diff
@@ -0,0 +1,59 @@
+puts "How many employees will be processed?" 
+employees = gets.chomp.to_i #convert for arithmetic
+
+while employees > 0 
+	puts "What is your name?" # Question 1
+	name = gets.chomp
+	puts "How old are you?"   # Question 2
+	age = gets.chomp.to_i
+	puts "What year were you born?" # Question 3
+	yearmatch = true
+	byear = gets.chomp.to_i
+		if (2016-byear) == age # roughly check age by using current year
+			yearmatch # yearmatch is true still
+		else
+			yearmatch = false
+		end
+	puts "Our company cafeteria serves garlic bread. 
+	Should we order some for you?" # Question 4
+		order_g = gets.chomp
+		if order_g == "yes".upcase || order_g == "yes".capitalize! || order_g == "yes"
+			order_g = true #set to true
+		else 
+			order_g = false
+		end
+
+	puts "Would you like to enroll in the company's health insurance?" #Question 5
+		health = gets.chomp
+		if health == "yes".upcase || health == "yes".capitalize! || health == "yes"
+			health = true
+		else 
+			health = false
+		end
+	allergy = "" # Allergy variable, made accessible outside of loop for case statement
+	loop do
+		puts "List any allergies. Type 'done' when finished."
+		allergy = gets.chomp
+		if allergy == "sunshine" || allergy == "done"
+			break
+		end 
+	end
+
+	case # case statements
+	when (!yearmatch && (!order_g || !health))|| allergy == "sunshine"
+		puts "Probably a vampire."

```

Comment:
puts is an interesting method, it takes an object as an argument, displays it to $STDOUT (your terminal), and returns nil. The thing to remember with puts is that it returns nil after it is called, so when we write code like
def add
	puts 1 + 2
end
we think we are returning 3 when we call the method because it displays to our screen, but we actually get nil (as the the last evaluated expression is the return of a method).
To avoid this, just omit puts from the method. Now you may say, 'But how do we know it's working?' Great question! For starters we could call our method and display its return with the more friendly development-only method 'p' like so.
	p add
The gist is, only use puts when you want to display text to the screen, if your methods are supposed to do something else, let them and if you are curious look at the results outside of that method. In this case, I'd suggest using a variable like the previous comment `results = 'some string'` so you can reset that variable according to whatever conditions are met.